vortex-array/src/array/byte_bool/mod.rs

```diff
@@ -0,0 +1,161 @@
+use std::mem::ManuallyDrop;
+
+use arrow_buffer::{BooleanBuffer, NullBuffer};
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+use vortex_buffer::Buffer;
+
+use super::bool::BoolArray;
+use crate::{
+    impl_encoding,
+    validity::{ArrayValidity, LogicalValidity, Validity, ValidityMetadata},
+    visitor::{AcceptArrayVisitor, ArrayVisitor},
+    ArrayFlatten,
+};
+
+mod compute;
+mod stats;
+
+impl_encoding!("vortex.byte_bool", ByteBool);
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ByteBoolMetadata {
+    validity: ValidityMetadata,
+    length: usize,
+}
+
+impl ByteBoolArray {
+    pub fn validity(&self) -> Validity {
+        self.metadata()
+            .validity
+            .to_validity(self.array().child(0, &Validity::DTYPE))
+    }
+
+    pub fn try_with_validity<V: Into<Validity>>(
+        data: Vec<bool>,
+        validity: V,
+    ) -> VortexResult<Self> {
+        let validity = validity.into();
+        let mut vec = ManuallyDrop::new(data);
+        vec.shrink_to_fit();
+
+        let ptr = vec.as_mut_ptr() as *mut u8;
+        let length = vec.len();
+        let capacity = vec.capacity();
+
+        let bytes = unsafe { Vec::from_raw_parts(ptr, length, capacity) };
+
+        let buffer = Buffer::from(bytes);
+        let typed = TypedArray::try_from_parts(
+            DType::Bool(validity.nullability()),
+            ByteBoolMetadata {
+                validity: validity.to_metadata(length)?,
+                length,
+            },
+            Some(buffer),
+            validity.into_array_data().into_iter().collect_vec().into(),
```

Comment:
I think this should just be `vec![validity.into_array_data()]`?

Reply:
No, took this snippet from `BoolArray`, it can probably be simplified but I suspect that's going to take a bigger change